### PR TITLE
Added support for fetch-incidents in return_error

### PR DESCRIPTION
**Pull request by DeanArbel**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/16998

## Description
Added support for fetch-incidents in return_error (if command is fetch-incidents will throw an exception instead of returning error)

## Does it break backward compatibility?
   - No

## Must have
- [x] Tests
- [ ] Code Review

## Additional changes
Handled potential bug in case error message contained non-ascii character

### CODE REVIEW

**Comment by bakatzir** [comment]
@DeanArbel  @jochman please sync: https://github.com/demisto/content/pull/4069

**Comment by DeanArbel** [comment]
@orenzohar mentioned there was an open PR before that was not merged due to a reason he does not remember.
@ShahafBenYakir @ronykoz do you know of that?

**Comment by ronykoz** [comment]
I think it was something that Oren implemented in the old commonServerPython and I opened a PR for moving it to the package and then Oren closed it for some reason right before I went to the surgery.
Lets talk about it tomorrow when I’m in the office

**Comment by ShahafBenYakir** [comment]
@DeanArbel @ronykoz This is the PRhttps://github.com/demisto/content/pull/3321 and the reason it was closed is that it did not solve the issue. The solution should’ve been in common server Like Dean implemented.

File: Scripts/CommonServerPython/CommonServerPython_test.py [hunk 1]
```diff
@@ -482,3 +483,39 @@ def test_is_mac_address():
...
+    }
+
+    # Test command that is not fetch-incidents
+    mocker.patch.object(demisto, 'command', return_value="test-command")
+    mocker.patch.object(sys, 'exit')
+    mocker.spy(demisto, 'results')
+    return_error(err_msg, '', outputs)
+    assert str(demisto.results.call_args) == "call({})".format(expected_error)
+
+    # Test fetch-incidents
```
Comment (ronykoz) [inline Scripts/CommonServerPython/CommonServerPython_test.py]: Better to separate this one to another test, each test module in python should test on flow of functionality - thats the best practice

File: Scripts/CommonServerPython/CommonServerPython.py [hunk 2]
```diff
@@ -1372,13 +1372,18 @@ def return_error(message, error='', outputs=None):
...
     LOG.print_log()
-    demisto.results({
-        'Type': entryTypes['error'],
-        'ContentsFormat': formats['text'],
-        'Contents': str(message),
-        "EntryContext": outputs
-    })
-    sys.exit(0)
+    if not isinstance(message, str):
+        message = message.encode('utf8') if hasattr(message, 'encode') else str(message)
```
Comment (ronykoz) [inline Scripts/CommonServerPython/CommonServerPython.py]: please add a blank line afterwards for a logical separation

File: Scripts/CommonServerPython/CommonServerPython.py [hunk 3]
```diff
@@ -1372,13 +1372,19 @@ def return_error(message, error='', outputs=None):
...
+
+    if hasattr(demisto, 'command') and demisto.command() == 'fetch-incidents':
+        raise Exception(message)
+    else:
+        demisto.results({
+            'Type': entryTypes['error'],
+            'ContentsFormat': formats['text'],
+            'Contents': message,
+            "EntryContext": outputs
```
Comment (Itay4) [inline Scripts/CommonServerPython/CommonServerPython.py]: need to be consistent here, `"` -> `'`

**Comment by yaakovi** [comment]
forced since it run all tests and failed. (relevant steps finished successfully)

**Pull request merged**